gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenFragment.java

```diff
@@ -232,10 +236,25 @@ private void closeDrawer() {
   }
 
   @Override
-  public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+  public void onCreateOptionsMenu(@NonNull Menu menu, MenuInflater inflater) {
     inflater.inflate(R.menu.feature_sheet_menu, menu);
   }
 
+  @Override
+  public boolean onOptionsItemSelected(@NonNull MenuItem item) {
+    switch (item.getItemId()) {
+      case R.id.move_feature_menu_item:
+        // TODO
+        return false;
+      case R.id.delete_feature_menu_item:
+        // TODO: Re-position map to default location after successful deletion.
+        viewModel.deleteActiveFeature().as(autoDisposable(this)).subscribe(this::hideBottomSheet);
```

Comment:
With `autoDisposable()` used here, I believe it will scope the subscription to the entire lifecycle of the HomeScreenFragment. These will accumulate as long as the home screen remains open.

To fix this, you can create a `PublishProcessor deleteFeatureRequests` in the VM which you could populate with feature id on each deletion request. That stream would then be flatMapped to the actual feature deletion observable, the resulting stream exposed as a stream of SingleEvent that is subscribed to in one of the `create` methods of this class. Wdyt?

Reply:
Done. Thanks